clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -871,15 +896,11 @@ invariant_checker_t::check_for_pc_discontinuity(
             TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
                     shard->file_type_) ||
             // Regular fall-through.
-            (prev_instr_trace_pc + shard->prev_instr_.instr.size == memref.instr.addr) ||
-            // String loop.
-            (prev_instr_trace_pc == memref.instr.addr &&
-             (memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH ||
-              // Online incorrectly marks the 1st string instr across a thread
-              // switch as fetched.
-              // TODO i#4915, #4948: Eliminate non-fetched and remove the
-              // underlying instrs altogether, which would fix this for us.
-              (!knob_offline_ && shard->saw_timestamp_but_no_instr_))) ||
```

Comment:
For the prev_xfer_marker relaxation below:

        // TODO i#5912: Isn't this relaxation too loose since we really only want
        // to relax if a kernel event happened immediately prior, while prev_xfer_marker_
        // could be many records back?